api/src/main/java/org/openmrs/api/context/UserContext.java

```diff
@@ -96,8 +100,15 @@ public User authenticate(String username, String password, ContextDAO contextDAO
 		if (log.isDebugEnabled()) {
 			log.debug("Authenticating with username: " + username);
 		}
-		
-		this.user = contextDAO.authenticate(username, password);
+		try {
+			this.user = contextDAO.authenticate(username, password);
+			notifyUserSessionListener(this.user, Event.LOGIN, Status.SUCCESS);
+		} catch(ContextAuthenticationException e) {
+			User loggingInUser = new User();
```

Comment:
Actually, I couldn't understand this method. Why you initiate a new user when the `contextDAO.authenticate()` throws a `ContextAuthenticationException` ?